src/theme/Navbar/Content/index.tsx

```diff
@@ -43,27 +53,143 @@ const NavbarContentLayout = ({left, right}) => {
   )
 }
 
+// Custom search component
+const CustomSearch = () => {
+  const [showSearchIcon, setShowSearchIcon] = useState<boolean>(false)
+  const [isSearchModalVisible, setIsSearchModalVisible] = useState<boolean>(false)
+  const history = useHistory()
+
+  // Handlers to control search visibility
+  function handleSearchClick() {
+    setIsSearchModalVisible(true)
+  }
+
+  function handleSearchModalClose() {
+    setIsSearchModalVisible(false)
+  }
+
+  // Function to handle zoom behavior based on input focus
+  const handleZoomBehavior = () => {
+    const viewportMetaTag = document.querySelector('meta[name="viewport"]') as HTMLMetaElement
+
+    if (viewportMetaTag) {
+      // Enable user zooming when no input is in focus
+      viewportMetaTag.content = "width=device-width, initial-scale=1, maximum-scale=5, user-scalable=yes"
+
+      // Add an event listener to detect when an input is in focus
+      document.addEventListener("focusin", handleInputFocus)
+
+      // Remove the event listener when the component unmounts or the modal closes
+      return () => {
+        document.removeEventListener("focusin", handleInputFocus)
+      }
+    }
+  }
+
+  // Function to handle input focus
+  const handleInputFocus = (event: FocusEvent) => {
+    const isInput = (event.target as HTMLElement).tagName.toLowerCase() === "input"
+
+    // Disable user zooming when an input is in focus
+    const viewportMetaTag = document.querySelector('meta[name="viewport"]') as HTMLMetaElement
+    if (viewportMetaTag && isInput) {
+      viewportMetaTag.content = "width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no"
+    }
+  }
+
+  useEffect(() => {
+    // Check pathname for showing the search icon
+    window.location.pathname.includes("/docs/") ? setShowSearchIcon(true) : setShowSearchIcon(false)
```

Comment:
https://docusaurus.io/docs/advanced/routing refer

Reply:
using the useLocation hook from docusaurus